### PR TITLE
fix: Use dynamic import for VoiceRecorder

### PR DESCRIPTION
**Pull request by HoonBaek**
### Description Of Changes

* Use dynamic import to lazy load the audio processor only when using the voice message feature

[UIKIT-3590](https://sendbird.atlassian.net/browse/UIKIT-3590)

[UIKIT-3590]: https://sendbird.atlassian.net/browse/UIKIT-3590?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/hooks/VoiceRecorder/index.tsx [hunk 1]
```diff
@@ -34,11 +33,25 @@ const VoiceRecorderContext = createContext<VoiceRecorderContext>({
...
   const [isRecordable, setIsRecordable] = useState<boolean>(false);
 
+  const [webAudioUtils, setWebAudioUtils] = useState(null);
+  useEffect(() => {
+    if (isVoiceMessageEnabled && !webAudioUtils) {
+      import('./WebAudioUtils').then((data) => {
+        setWebAudioUtils(data);
+      })
+    }
+  }, []);
```
Comment (AhyoungRyu) [inline src/hooks/VoiceRecorder/index.tsx:47]: Maybe we can have a separate hook like 

```
const useLazyLoading({
  cond,
  module,
}) {
  const [data, setData] = useState(null);
  useEffect(() => {
    if (cond) {
      import(module).then((data) => {
        setData(data);
      })
    }
  }, []);
  
  return null;
}
```

and its usage would be like 
```
useLazyLoading({ cond: ..., module: ... });
```

**Review by AhyoungRyu — APPROVED** [review]

File: src/hooks/VoiceRecorder/index.tsx (hunk 1 above)
Comment (HoonBaek) [reply src/hooks/VoiceRecorder/index.tsx:47]: Yeah, that's a good idea, let me keep this topic to the next sprint

File: src/hooks/VoiceRecorder/index.tsx [hunk 2]
```diff
@@ -34,11 +33,25 @@ const VoiceRecorderContext = createContext<VoiceRecorderContext>({
 export const VoiceRecorderProvider = (props: VoiceRecorderProps): React.ReactElement => {
   const { children } = props;
   const { config } = useSendbirdStateContext();
-  const { logger } = config;
+  const { logger, isVoiceMessageEnabled } = config;
   const [mediaRecorder, setMediaRecorder] = useState<MediaRecorder>(null);
   const [isRecordable, setIsRecordable] = useState<boolean>(false);
 
+  const [webAudioUtils, setWebAudioUtils] = useState(null);
```
Comment (sravan-s) [inline src/hooks/VoiceRecorder/index.tsx:40]: Hmm, here, maybe it was better to have a simple audioUtil, that doesn't convert file to mp3

Comment (sravan-s) [reply src/hooks/VoiceRecorder/index.tsx:40]: cc @HoonBaek

**Comment by HoonBaek** [comment]
image

**Pull request merged**